src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerJsonProcessor.java

```diff
@@ -239,6 +243,94 @@ public Content buildProviderJson(final Repository repository,
         ContentTypes.APPLICATION_JSON));
   }
 
+  /**
+   * Merges an incoming set of packages.json files.
+   */
+  public Content mergePackagesJson(final Repository repository, final List<Payload> payloads) throws IOException {
+    Set<String> names = new HashSet<>();
+    for (Payload payload : payloads) {
```

Comment:
Trivial: I think you could consider using functional approach here

Reply:
> Trivial: I think you could consider using functional approach here

The reason I didn't do that was because we'd end up having to go the route of using `UncheckedIOException` because of the difficulty of throwing checked exceptions from within the lambdas in the stream.